### PR TITLE
HERA-162 - [Development] [LMS part]Create new XBlock "End Survey"

### PR DESCRIPTION
**Pull request by sendr**
**Description:** [Development] [LMS part]Create new XBlock "End Survey"

**Youtrack: HERA-162** https://youtrack.raccoongang.com/issue/HERA-162

**Reviewers:**
- [ ] @sendr 
- [ ] @UvgenGen 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation in source code updated
- [ ] All related Confluence documentation is updated (including deployment documentation)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import pkg_resources
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope, String, List
+from xblock.fields import JSONField
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblockutils.resources import ResourceLoader
+from xblock.runtime import IdGenerator, IdReader, KeyValueStore, KvsFieldData, NoSuchViewError, NullI18nService, Runtime
```
Comment (UvgenGen) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: Do you use all of these imports?

File: hera_xblocks/end_survey/end_survey/static/css/end_survey.css [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+  
+  .unit-title {
+      display: none;
+  }
+  .row {
+    margin: 0 0 20px;
+    display: flex;
+    justify-content: space-between;
+  }
```
Comment (UvgenGen) [inline hera_xblocks/end_survey/end_survey/static/css/end_survey.css]: empty line

File: hera_xblocks/end_survey/end_survey/end_survey.py (hunk 1 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: Deleted not used one

File: hera_xblocks/end_survey/end_survey/static/css/end_survey.css (hunk 2 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/static/css/end_survey.css]: Added line

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+@XBlock.wants('user')
+class EndSurveyXBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    XBLock response to take the surver from the student.
+    When he passed the lesson.
+    """
+
+    editable_fields = ('data',)
+    data = JSONField(default={})
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: regroup xblock fields and xblock settings attributes

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        """
+        html = self.user_is_staff()
+        frag = Fragment(html)
+        frag.add_css(self.resource_string("static/css/end_survey.css"))
+        frag.add_javascript(self.resource_string("static/js/src/end_survey.js"))
+        frag.add_javascript_url("https://cdn.jsdelivr.net/gh/vast-engineering/jquery-popup-overlay@2/jquery.popupoverlay.min.js")
+
+        frag.initialize_js('EndSurveyXBlock')
+        return frag
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: add empty line

File: hera_xblocks/end_survey/end_survey/static/css/end_survey.css [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/static/css/end_survey.css]: lack of empty line

File: hera_xblocks/end_survey/end_survey/static/js/src/end_survey.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+/* Javascript for EndSurveyXBlock. */
+function EndSurveyXBlock(runtime, element) {
+    let handlerUrl = runtime.handlerUrl(element, 'users_statistic')
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/static/js/src/end_survey.js]: trailing semicolon

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+from web_fragments.fragment import Fragment
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope, String, List
+from xblock.fields import JSONField
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+from xblockutils.resources import ResourceLoader
+from xblock.runtime import IdGenerator, IdReader, KeyValueStore, KvsFieldData, NoSuchViewError, NullI18nService, Runtime
+from xblock.reference.user_service import XBlockUser, UserService
+from django.contrib.auth.models import User
+from student.models import anonymous_id_for_user, get_user_by_username_or_email
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: Reordered

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    def questions(self):
+        return self.data.get("questions")
+
+    @property
+    def title(self):
+        return self.data.get("title")
+
+    def user_is_staff(self):
+        """ Checking user is staff or not
+        and return the different html for the student and for the staff """
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: Done

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+    # has_author_view = True  # Without this flag, studio will use student_view on newly-added blocks :/
+
+    editable_fields = ('data',)
+    data = JSONField(default={})
+
+    user_count = Integer(
+        default=0, scope=Scope.user_state_summary,
+        help="A simple counter, to show something happening",
+    )
+    user_reusult = String(scope=Scope.user_state)
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: Done

File: hera_xblocks/end_survey/end_survey/end_survey.py (hunk 4 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: done

File: hera_xblocks/end_survey/end_survey/static/css/end_survey.css (hunk 5 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/static/css/end_survey.css]: done

File: hera_xblocks/end_survey/end_survey/static/js/src/end_survey.js (hunk 6 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/static/js/src/end_survey.js]: done

File: hera_xblocks/end_survey/end_survey/end_survey.py (hunk 3 above)
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: done

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+        html = loader.render_django_template(
+            template,
+            context=context
+        )
+
+        return html
+
+    def get_context(self):
+        return {
+            "last_choice": self.last_choice,
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: Changed to 'user_result'

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+    @XBlock.json_handler
+    def get_data(self, somedata, sufix=''):
+        return self.data
+
+    @XBlock.json_handler
+    def users_statistic(self, data, sufix=''):
+        """counting amount of students who answer on the survey"""
+
+        self.user_reusult = data
```
[the remark this answers is not in the document]
Comment (RuslanSkiraRaccoongang) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: 'reusult'
 changed to 'result'

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+        
+        html = self.user_is_staff()
+        frag = Fragment(html)
+        frag.add_css(self.resource_string("static/css/end_survey.css"))
+        frag.add_javascript(self.resource_string("static/js/src/end_survey.js"))
+        frag.add_javascript_url("https://cdn.jsdelivr.net/gh/vast-engineering/jquery-popup-overlay@2/jquery.popupoverlay.min.js")
+
+        frag.initialize_js('EndSurveyXBlock')
+        return frag
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: empty line

File: hera_xblocks/end_survey/end_survey/end_survey.py (hunk 3 above)
Comment (sendr) [reply hera_xblocks/end_survey/end_survey/end_survey.py]: not done, separate please fields from xblock settings properties

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+    user_count = Integer(default=0, scope=Scope.user_state_summary)
+    result_summary = List(default=[], scope=Scope.user_state_summary)
+    user_result = List(default=[], scope=Scope.user_state_summary)
+
+
+    @property
+    def questions(self):
+        return self.data.get("questions")
+
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: should be only one line

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+    @property
+    def title(self):
+        return self.data.get("title")
+
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: the same above

File: hera_xblocks/end_survey/end_survey/end_survey.py [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+        return {
+            "questions": self.questions,
+            "title": self.title,
+            "user_count": self.user_count
+            # TODO: wrire self.result_summary
+            # "result_summary": ''
+        }
+
+
```
Comment (sendr) [inline hera_xblocks/end_survey/end_survey/end_survey.py]: check please all places and replace them to one line

**Pull request merged**